File: gamsojji/backjoon/step6/2444.js
let fs = require("fs");
const filePath = process.platform === 'linux' ? "/dev/stdin" : `${__dirname}/input.txt`;
let num = fs.readFileSync(filePath).toString().trim();
num = parseInt(num);
let result = "";
for (let i = 1; i <= num; i++) {
    
    for(let j = 0; j < num-i; j++){
        result+= " ";
    }
    for(let k = 0; k < 1+2*(i-1); k++){
        result+= "*";
    }
    result+="\n"
}
for (let i = num-1; i > 0; i--) {
    
    for(let j = 0; j < num-i; j++){
        result+= " ";
    }
    for(let k = 0; k < 1+2*(i-1); k++){
        result+= "*";
    }
    if(i>1){
        result+="\n"
    }
}
console.log(result);

File: rolemadelen/baekjoon/6단계/2444.rs
use std::io::stdin;

fn main() {
    let mut input = String::new();
    stdin().read_line(&mut input).unwrap();
    let n = input.trim().parse().unwrap();

    for i in 0..n {
        for j in 0..n-i-1 {
            print!(" ");
        }
        for j in 0..(2*i)+1 {
            print!("*");
        }
        println!();
    }
    for i in (0..n-1).rev() {
        for j in 0..n-i-1 {
            print!(" ");
        }
        for j in 0..(2*i)+1{
            print!("*");
        }
        println!();
    }
}
